### PR TITLE
Add documentations for MLflow authentication

### PR DESCRIPTION
**Pull request by gabrielfu**
## Related Issues/PRs

#724 #8130 #8286 #8548 

## What changes are proposed in this pull request?

Add documentations for MLflow authentication, AuthServiceClient and newly added authentication REST API endpoints

## How is this patch tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Manually checked the generated docs.

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [X] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: docs/source/auth.rst
Comment (gabrielfu) [inline docs/source/auth.rst:128]: a few endpoints including "search experiments" are not present in the REST API page, so the ref is not working

Comment (harupy) [reply docs/source/auth.rst]: We actually found `rest-api.rst` is outdated. Should be fixed soon.

File: docs/source/rest-api.rst
Comment (harupy) [inline docs/source/rest-api.rst:1800]: Is it possible to isolate these endpoints into a different page?

Comment (harupy) [reply docs/source/rest-api.rst:1800]: Feel free to create a folder for auth and put relevant files there.

Comment (gabrielfu) [reply docs/source/rest-api.rst:1800]: sure, sounds good!

File: docs/source/auth.rst
Comment (gabrielfu) [reply docs/source/auth.rst]: that's great! i'll keep an eye out

File: docs/source/_static/images/auth_signup.png
Comment (harupy) [inline docs/source/_static/images/auth_signup.png:1]: Can we rename file this to `auth_signup_form.png`?

File: docs/source/_static/images/auth_ui.png
Comment (harupy) [inline docs/source/_static/images/auth_ui.png:1]: I think `auth_prompt.png` is more accurate name.

File: mlflow/server/auth/client.py
Comment (harupy) [inline mlflow/server/auth/client.py:24]: Is it possible to make `mlflow.server.get_app_client()` a link?

File: docs/source/auth.rst
Comment (harupy) [reply docs/source/auth.rst]: Merged! Can you rebase?

File: docs/source/auth/index.rst
Comment (harupy) [inline docs/source/auth/index.rst:1]: This looks fantastic!

File: docs/source/auth/index.rst
Comment (harupy) [inline docs/source/auth/index.rst:525]:
```suggestion
        "https://<mlflow_tracking_uri>/api/2.0/mlflow/users/create",
```

Can we add `<>` here to clarify this is a placeholder that needs to be replaced?

File: docs/source/auth/index.rst
Comment (harupy) [reply docs/source/auth/index.rst:1]: I'm going to file a PR to allow setting up credentials in `~/.mlflow/credentials` :) We might need a small adjustment once that's merged.

Comment (harupy) [reply docs/source/auth/index.rst:1]: Opened #8688

File: docs/source/_static/images/auth_signup.png
Comment (gabrielfu) [reply docs/source/_static/images/auth_signup.png:1]: sure!

File: docs/source/_static/images/auth_ui.png
Comment (gabrielfu) [reply docs/source/_static/images/auth_ui.png:1]: sure

File: mlflow/server/auth/client.py
Comment (gabrielfu) [reply mlflow/server/auth/client.py:24]: sure, i added the docs for it

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst:1]: thanks! :) I'll add the missing part

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst:525]: sure!

File: docs/source/auth.rst
Comment (gabrielfu) [reply docs/source/auth.rst]: great! i added them back

File: docs/source/auth/rest-api.rst
Comment (harupy) [inline docs/source/auth/rest-api.rst:7]: Can we add a note and say this is an experimental feature?

File: docs/source/auth/index.rst
Comment (harupy) [inline docs/source/auth/index.rst:467]: Can we modify the order of sub-sections in this section?

- Environment variables
- Credentials File
- REST API

File: docs/source/auth/index.rst
Comment (harupy) [inline docs/source/auth/index.rst:89]:
```suggestion
     - Required permission
```

for clarity.

File: docs/source/auth/index.rst
Comment (harupy) [inline docs/source/auth/index.rst:97]: If we split the table, can we remove `ExperimentPermission`?

**Review by harupy — APPROVED** [review]
Left a few more comments, LGTM once they are addressed!

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:387]:
```suggestion
granting or revoking permissions from other users, and managing permissions for all 
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:388]:
```suggestion
MLflow resources,** even if ``NO_PERMISSIONS`` is explicitly set to that admin account.
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:390]:
```suggestion
MLflow has a built-in admin user that will be created the first time that the MLflow authentication feature is enabled.
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:392]:
```suggestion

.. note::
    It is recommended that you update the password as soon as possible after it is created.

The default admin user credentials are as follows:

```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:462]:
```suggestion
Currently, MLflow UI does not display any information about the current user.
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:505]:
```suggestion
You can save your credentials in a file to remove the need for setting environment variables every time.
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:506]:
```suggestion
The credentials should be saved in ``~/.mlflow/credentials`` using ``INI`` format. Note that the password 
```

File: docs/source/auth/index.rst
Comment (BenWilson2) [inline docs/source/auth/index.rst:520]:
```suggestion
Creating a New User
```

File: mlflow/server/auth/client.py
Comment (BenWilson2) [inline mlflow/server/auth/client.py:162]: Can this be changed as well as the .rst docs so that we're not suggesting users write down their username and password in a script in plaintext?

File: mlflow/server/auth/client.py
Comment (BenWilson2) [inline mlflow/server/auth/client.py:190]: ditto

File: mlflow/server/auth/client.py
Comment (BenWilson2) [inline mlflow/server/auth/client.py:225]: and here as well

File: mlflow/server/auth/client.py
Comment (BenWilson2) [inline mlflow/server/auth/client.py:229]: this should also reference an environment variable that is set on the system

File: mlflow/server/auth/client.py
Comment (BenWilson2) [inline mlflow/server/auth/client.py:268]: environment variable instead of plaintext

File: docs/source/auth/rest-api.rst
Comment (gabrielfu) [reply docs/source/auth/rest-api.rst:7]: sure

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst:471]: okay

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst:500]: updated

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst:80]: sure

File: mlflow/server/auth/client.py
Comment (gabrielfu) [reply mlflow/server/auth/client.py]: makes sense, I changed them

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks so much, @gabrielfu !

File: docs/source/auth/index.rst
Comment (gabrielfu) [reply docs/source/auth/index.rst]: i have changed the section a bit

**Comment by gabrielfu** [comment]
> LGTM! Thanks so much, @gabrielfu !

No problem! :)

**Comment by dstack4273** [comment]
These changes look awesome! How long does it usually take for a big change like this to be cut as part of the release?

**Pull request merged**
